sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/aio/_data_lake_directory_client_async.py

```diff
@@ -692,7 +694,7 @@ def get_paths(
         :returns: An iterable (auto-paging) response of PathProperties.
         :rtype: ~azure.core.async_paging.AsyncItemPaged[~azure.storage.filedatalake.PathProperties]
         """
-        hostname = self._hosts[self._location_mode]
+        hostname = self._hosts[self._location_mode]  # type: ignore [index]
```

Comment:
Just leaving another comment on `_hosts` to show its in multiple places in sync/async.

Reply:
Thanks! Please see comments above for the constructor.